### PR TITLE
Update GetTrades and GetContractors endpoints

### PR DESCRIPTION
**Pull request by LBHCallumM**
Update GetTrades and GetContractors endpoints to accept optional parameters. However, if `GetAllContractors` or `GetAllTrades` is false, the other parameters are still required.

### CODE REVIEW

File: RepairsApi/V2/Controllers/ContractorController.cs [hunk 1]
```diff
@@ -22,21 +24,29 @@ public ContractorController(IScheduleOfRatesGateway sorGateway)
...
-            var contractors = await _sorGateway.GetContractors(propertyReference, tradeCode);
+            if (!request.GetAllContractors && string.IsNullOrWhiteSpace(request.TradeCode))
+            {
+                return BadRequest($"If {nameof(request.GetAllContractors)} is False, you must include a value for {nameof(request.TradeCode)}");
+            }
 
-            return Ok(contractors);
+            if (!request.GetAllContractors && string.IsNullOrWhiteSpace(request.PropertyReference))
+            {
+                return BadRequest($"If {nameof(request.GetAllContractors)} is False, you must include a value for {nameof(request.PropertyReference)}");
```
Comment (Asura5) [inline RepairsApi/V2/Controllers/ContractorController.cs:41]: What if they include both?

File: RepairsApi/V2/Controllers/ScheduleOfRatesController.cs [hunk 2]
```diff
@@ -103,12 +103,21 @@ public async Task<IActionResult> GetSorCode([Required] string sorCode, [FromQuer
...
+        [Route("trades")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(IEnumerable<SorTradeResponse>), StatusCodes.Status200OK)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> ListTrades(ListTradesRequestObject request)
         {
-            return Ok(await _listSorTrades.Execute(propRef));
+            if (!request.GetAllTrades && string.IsNullOrWhiteSpace(request.PropertyReference))
+            {
+                return BadRequest($"If {nameof(request.GetAllTrades)} is False, you must include a value for {nameof(request.PropertyReference)}");
```
Comment (Asura5) [inline RepairsApi/V2/Controllers/ScheduleOfRatesController.cs:115]: What if they include both?

File: RepairsApi/V2/Controllers/ContractorController.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi/V2/Controllers/ContractorController.cs:41]: Both as in propRef and getAllContractors? GetAllContractors returns all contractors, so it doesn't need the propRef. Its the same for trades

Comment (Asura5) [reply RepairsApi/V2/Controllers/ContractorController.cs:41]: So why is there not validation disallowing a request where both of them are supplied

**Pull request merged**